Mathlib/Analysis/SpecialFunctions/NthRoot.lean

```diff
@@ -0,0 +1,203 @@
+import Mathlib.Analysis.SpecialFunctions.Pow.Real
+
+/-!
+# nth root operations
+
+This file provides `Real.nthRoot n` to compute `ⁿ√`,
+which operates as expected on negative values when `n` is odd.
+The trap this avoids is that using `rpow`, `(-8 : ℝ) ^ (1/3 : ℝ) = 1`.
+
+-/
+
+noncomputable section
+
+namespace Real
+
+section qpow
+
+instance instQPow : Pow ℝ ℚ where
+  pow r q :=
+    if Even q.den then r ^ (q : ℝ) else SignType.sign r ^ (q.den * q.num) * abs r ^ (q : ℝ)
+
+theorem qpow_of_even (r : ℝ) {q : ℚ} (hn : Even q.den) : r ^ q = r ^ (q : ℝ) :=
+  if_pos hn
+
+theorem qpow_of_den_eq_one (r : ℝ) {q : ℚ} (hq : q.den = 1) : r ^ q = r ^ q.num := by
+  refine if_neg (by simp [hq]) |>.trans ?_
+  rw [hq, Nat.cast_one, one_mul]
+  conv_lhs => enter [2]; rw [← Rat.coe_int_num_of_den_eq_one hq, Rat.cast_intCast, rpow_intCast]
+  rw [← mul_zpow, sign_mul_abs]
+
+theorem qpow_of_odd (r : ℝ) {q : ℚ} (hn : Odd q.den) :
+    r ^ q = SignType.sign r ^ (q.den * q.num) * abs r ^ (q : ℝ) :=
+  if_neg <| Nat.not_even_iff_odd.mpr hn
+
+@[simp, norm_cast]
+theorem qpow_intCast (r : ℝ) (z : ℤ) : r ^ (z : ℚ) = r ^ z :=
+  qpow_of_den_eq_one _ <| Rat.den_intCast _
+
+@[simp, norm_cast]
+theorem qpow_natCast (r : ℝ) (n : ℕ) : r ^ (n : ℚ) = r ^ n :=
+  qpow_intCast _ _
+
+@[simp]
+theorem qpow_ofNat (r : ℝ) (n : ℕ) : r ^ (ofNat(n) : ℚ) = r ^ ofNat(n) :=
+  qpow_natCast _ _
+
+theorem qpow_zero (r : ℝ) : r ^ (0 : ℚ) = 1 := by
+  rw [qpow_ofNat, pow_zero]
+
+theorem qpow_one (r : ℝ) : r ^ (1 : ℚ) = r := by
+  rw [qpow_ofNat, pow_one]
+
+theorem qpow_of_odd_of_nonpos {q : ℚ} (hn : Odd q.den) {r : ℝ} (hr : r ≤ 0) :
+    r ^ q = (-1) ^ (q.den * q.num) * (-r) ^ (q : ℝ) := by
+  rw [qpow_of_odd _ hn]
+  obtain rfl | hr := hr.eq_or_lt
+  · simp
+    obtain rfl | hq := eq_or_ne q 0
+    · simp
+    have hn0 : q ≠ 0 := sorry
+    rw [zero_rpow (mod_cast hn0)]
+  · rw [abs_of_neg hr, _root_.sign_neg hr]
+    simp
+
+theorem qpow_of_nonneg {q : ℚ} {r : ℝ} (hr : 0 ≤ r) :
+    r ^ q = r ^ (q : ℝ) := by
```

Comment:
A lemma stating `r ^ q = r ^ (q : ℝ) \iff 0 \le r \or Even q.num \or Even q.den` (or something similar) would be valuable.

Reply:
Done, though that isn't the true version of the statement.